junit/src/main/java/org/kaazing/k3po/junit/rules/Latch.java

```diff
@@ -129,19 +132,18 @@ boolean hasException() {
 
     void notifyException(Exception exception) {
         this.exception = exception;
-        // Commented because of issue https://github.com/k3po/k3po/issues/391
-//        prepared.countDown();
-//        startable.countDown();
-//        finished.countDown();
-        if (testThread != null) {
-            if (testThread != Thread.currentThread())
-                testThread.interrupt();
-        }
+        interruptTestThread();
```

Comment:
I don't understand how checking the current status will help. Can you please explain ?

This is the way it is working if we replace in the current code the `testThreadInterrupted` field with a call to `testThread.isInterrupted()` : the first interrupt will wake the main thread from the `awaitPrepared` or `awaitFinished` and will also clear the interrupted status. Thus, when the second one will come, `testThread.isInterrupted()` will show false, so it will set the status again. But this time, the main thread will not be in an await anymore, so the status will not be cleared.
It would be useful if we can check if the main thread is locked in an await. But I don't know if we can do that.